include/zephyr/acpi/acpi.h

```diff
@@ -173,6 +173,11 @@ int acpi_dmar_entry_get(enum AcpiDmarType type, ACPI_SUBTABLE_HEADER **tables);
 int acpi_drhd_get(enum AcpiDmarScopeType scope, ACPI_DMAR_DEVICE_SCOPE *dev_scope,
 		  union acpi_dmar_id *dmar_id, int *num_inst, int max_inst);
 
+typedef void (*dmar_foreach_func_t)(uintptr_t subtable);
```

Comment:
Why `uintptr_t` instead of e.g. `void *`? The previous `ACPI_DMAR_HEADER *` also seemed like a good approach, especially if `ACPI_DMAR_HEADER` is always the first element of each specialised DMAR struct - in that case you could use `CONTAINER_OF()` to get the actual type, without the need of any explicit or implicit type casts (which are always a bit risky in C since they will hide potential type conflicts).